tests/test_approximate.py

```diff
@@ -0,0 +1,489 @@
+# -*- coding: utf-8 -*-
```

Comment:
is this testing approx abstract class OR testing lsh? 
- Have you considered having test_approx and test_lsh separately? Would that make sense? 
- if new approx methods arrive, do we want them in this file?

Reply:
The _ApproximateNearest parent class code cannot run independently, similar to _Nearest (The test_nearest.py tests are for KNearest)

Reply:
test_nearest testing _Nearest makes perfect sense to me. Shall this be test_lsh then,?